### PR TITLE
[llvm-spirv][NFC] Add -emit-opaque-pointers to 2 llvm-spirv tests

### PR DESCRIPTION
**Pull request by MrSidims**

### CODE REVIEW

**Review by AlexeySachkov — APPROVED** [review]

File: llvm-spirv/test/transcoding/multiple_user_semantic_on_struct.ll [hunk 1]
```diff
@@ -25,13 +25,12 @@
...
 ; }
 
-
 ; RUN: llvm-as %s -o %t.bc
 ; RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
 
 ; RUN: llvm-spirv %t.bc -o %t.spv
-; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
-; RUN: llvm-dis -opaque-pointers < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+; RUN: llvm-spirv -r -opaque-pointers=1 %t.spv -o %t.rev.bc
```
Comment (bader) [inline llvm-spirv/test/transcoding/multiple_user_semantic_on_struct.ll:32]:
```suggestion
; RUN: llvm-spirv -r -emit-opaque-pointers %t.spv -o %t.rev.bc
```

File: llvm-spirv/test/transcoding/clk_event_t.cl [hunk 2]
```diff
@@ -2,11 +2,13 @@
...
 // RUN: spirv-val %t.spv
-// RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+// RUN: llvm-spirv -r -opaque-pointers=1 %t.spv -o %t.rev.bc
 // RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM-OCL
-// RUN: llvm-spirv -r --spirv-target-env=SPV-IR %t.spv -o %t.rev.bc
+// RUN: llvm-spirv -r -opaque-pointers=1 --spirv-target-env=SPV-IR %t.spv -o %t.rev.bc
 // RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM-SPV
 // RUN: llvm-spirv %t.rev.bc -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+//
+// XFAIL: *
```
Comment (bader) [inline llvm-spirv/test/transcoding/clk_event_t.cl:11]: This test still uses typed pointers, so I think we can use either XFAIL or `-opaque-pointers=1`, but not both of them at the same time.
BTW, the right name of the `llvm-spirv` option is `emit-opaque-pointers`. We should update CHECKs to use opaque pointers ASAP!

Comment (MrSidims) [reply llvm-spirv/test/transcoding/clk_event_t.cl:11]: IMO the test or translator is broken. It should generate target extension type, but it generates ptr. Tagging @jcranmer-intel

Comment (bader) [reply llvm-spirv/test/transcoding/clk_event_t.cl:11]: The test is definitely wrong - it doesn't check target extension type.
From your message it looks like another bug is in the clang.

**Pull request merged**